lib/src/wip/autocomplete/popup.dart

```diff
@@ -6,30 +6,43 @@ import 'popup_controller.dart';
 
 /// Popup window displaying the list of possible completions
 class Popup extends StatefulWidget {
-  final Offset normalOffset;
-  final Offset flippedOffset;
+  final PopupController controller;
   final Size editingWindowSize;
+
+  /// Represents the position of the editor widget in the coordinate system
+  /// of the application's window.
+  final Offset? editorOffset;
+
+  ///Represents the position of the popup window in the coordinate system
+  ///of the application's window when the popup is displayed above the editor.
```

Comment:
```suggestion
  /// The window coordinates of the highest allowed top-left corner
  /// of the popup if shown above the caret.
  ///
  /// Since the popup is pushed to the bottom of the allowed rectangle
  /// the actual position may be lower.
```